### PR TITLE
[APM][OTel] Ensure Errors views works with OTel Data

### PR DESCRIPTION
**Pull request by rmyz**
## Summary
Closes https://github.com/elastic/kibana/issues/192336

https://github.com/user-attachments/assets/7580ce62-74c5-4357-9add-a57b9bba63df

## How to test
1. Make sure to use https://github.com/elastic/otel-apm-e2e-poc and running with opentelemetry-demo data https://github.com/elastic/otel-apm-e2e-poc
2. Change the indices in Settings to only use `*otel*` indices.
3. Inside APM, search for `error.exception.handled:true` to see failing services
4. Deep dive into Errors and explore everything is working fine

### CODE REVIEW

File: x-pack/plugins/observability_solution/apm/server/routes/errors/get_error_groups/get_error_group_sample_ids.ts [hunk 1]
```diff
@@ -66,16 +67,18 @@ export async function getErrorGroupSampleIds({
           should: [{ term: { [TRANSACTION_SAMPLED]: true } }], // prefer error samples with related transactions
         },
       },
-      _source: [ERROR_ID, 'transaction'],
+      fields: [ERROR_ID],
```
Comment (MiriamAparicio) [inline x-pack/plugins/observability_solution/apm/server/routes/errors/get_error_groups/get_error_group_sample_ids.ts:70]: If I'm not mistaken, the 'transaction' was added for the scoring, so I think the field `TRANSACTION_SAMPLED`, should we also added not sure why just 'transaction' was in this case
@dgieselaar can you confirm?

File: x-pack/plugins/observability_solution/apm/server/routes/errors/get_error_groups/get_error_group_main_statistics.ts [hunk 2]
```diff
@@ -157,15 +159,17 @@ export async function getErrorGroupMainStatistics({
...
+        lastSeen: new Date(normalizedFields?.['@timestamp'] as string).getTime(),
         occurrences: bucket.doc_count,
-        culprit: bucket.sample.hits.hits[0]._source.error.culprit,
-        handled: bucket.sample.hits.hits[0]._source.error.exception?.[0].handled,
-        type: bucket.sample.hits.hits[0]._source.error.exception?.[0].type,
-        traceId: bucket.sample.hits.hits[0]._source.trace?.id,
+        culprit: normalizedFields?.error?.culprit,
+        handled: normalizedFields?.error?.exception[0]?.handled,
+        type: normalizedFields?.error?.exception[0]?.type,
+        traceId: normalizedFields?.trace?.id,
```
Comment (miloszmarcinkowski) [inline x-pack/plugins/observability_solution/apm/server/routes/errors/get_error_groups/get_error_group_main_statistics.ts:172]: nit: Optional chaining is only needed for `normalizedFields`. If the object is not undefined, then the rest is defined
```suggestion
        culprit: normalizedFields?.error.culprit,
        handled: normalizedFields?.error.exception[0].handled,
        type: normalizedFields?.error.exception[0].type,
        traceId: normalizedFields?.trace.id,
```

**Review by miloszmarcinkowski — APPROVED** [review]
LGTM 👍

**Comment by rmyz** [comment]
This work will be merged on https://github.com/elastic/kibana/pull/195242

**Pull request closed without merging**